### PR TITLE
Replace procedure syntax by explicit Unit annotation

### PR DESCRIPTION
**Pull request by pdalpra**
I realize this ends up as a huge PR, I can break it down to smaller pieces if you like.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by dwijnand** [comment]
Nice work. LGTM.

**Comment by pdalpra** [comment]
Fails to compile, though, made a typo...

File: compile/interface/src/main/scala/xsbt/DelegatingReporter.scala [hunk 1]
```diff
@@ -17,22 +17,21 @@ private object DelegatingReporter {
...
+  def error(msg: String): Unit = error(FakePos("scalac"), msg)
 
-  def printSummary() = delegate.printSummary()
+  def printSummary(): Unit = = delegate.printSummary()
 
   override def hasErrors = delegate.hasErrors
   override def hasWarnings = delegate.hasWarnings
   def problems = delegate.problems
-  override def comment(pos: Position, msg: String) = delegate.comment(convert(pos), msg)
+  override def comment(pos: Position, msg: String): Unit = = delegate.comment(convert(pos), msg)
```
Comment (dwijnand) [inline compile/interface/src/main/scala/xsbt/DelegatingReporter.scala]: typo, here and a little above

Comment (pdalpra) [inline compile/interface/src/main/scala/xsbt/DelegatingReporter.scala]: Yep, fixed them as you wrote that comment :)

**Comment by eed3si9n** [comment]
Thanks for this!

**Comment by eed3si9n** [comment]
I'm guessing that the Travis failures are just flaky tests, but I am restarting failed jobs to see if they pass.
LGTM

**Comment by pdalpra** [comment]
@eed3si9n Unfortunately, the'yre not, an issue on my side, fixing.

**Comment by pdalpra** [comment]
@eed3si9n The next PR build should be green.

**Comment by pdalpra** [comment]
Looks like it won't be green after all, but it's a "flaky" test:

```
[info] [error] (main/*:update) sbt.ResolveException: unresolved dependency: org.example#library-09d33fffabc50e24ba66_2.10;1.0-SNAPSHOT: not found
[info] [error] Total time: 2 s, completed Aug 4, 2015 8:26:39 AM
[error] x dependency-management / mvn-local
```

**Pull request merged**